zio-http/src/main/scala/zio/http/Channel.scala

```diff
@@ -5,6 +5,7 @@ import zio.{Task, Trace, UIO}
  * An immutable and type-safe representation of one or more netty channels. `A`
  * represents the type of messages that can be written on the channel.
  */
+// TODO Remove all netty-specific methods here and reduce footprint
```

Comment:
I think we can either restrict this trait or use a new, smaller one to accomplish the same goals here.